assets/scss/modules/_navigation.scss

```diff
@@ -147,3 +145,146 @@ a.back-to-top-link {
   }
 
 }
+
+/*
+Page Navigation
+
+A page navigation element with a previous and next link.
+
+Markup:
+<nav role="navigation" aria-label="Page navigation">
```

Comment:
So I first read about this here: http://html5doctor.com/using-aria-in-html/ which links to http://w3c.github.io/html/dom.html#aria-usage-note
But with discussing with @rpowis it seems like if there are non compliant screen readers then it may not hurt.
This is also an interesting read on this issue http://accessibleculture.org/articles/2010/05/html5-aria/

Sounds like we need to ask/test with Chris Moore